mirage/client/dns_client_mirage.ml

```diff
@@ -211,7 +211,7 @@ The format of a nameserver is:
       }
 
     let nameservers { proto ; nameservers ; _ } = proto, nameservers
-    let rng = R.generate ?g:None
+    let rng n = Cstruct.to_string (R.generate ?g:None n)
```

Comment:
And we can even could the middle man, and use `Mirage_crypto_rng_mirage.S`. The `mirage-random` should just vanish...

Reply:
I currently prefer to do this work in another PR. The implications of such a change, particularly as regards unikernels, have not been fully established. At this stage and in view of the release train, it's better to concentrate on a _releasable_ state rather than add another migration job.

Reply:
I only read your comment now, but pushed fdaa332f here. we can of course separate that commit into separate PR if you prefer.